lib/gems/pending/openstack/openstack_handle/metric_delegate.rb

```diff
@@ -17,6 +17,7 @@ def initialize(dobj, os_handle, name)
     def list_meters(filters)
       data = []
       filters.each do |filter|
+        filter['field'] = 'resource_id' if filter['field'] == 'metadata.resource_id'
         next if filter['field'] == 'metadata.instance_id'
         raise "Unexpected Metric filter \"#{filter['field']}\"" unless filter['field'] == 'resource_id'
```

Comment:
Out of curiosity, is there a reason why the change couldn't be to update the 'unless' check to also accept 'metadata.resource_id'?